stl/inc/xstring

```diff
@@ -2387,6 +2392,32 @@ struct _String_constructor_rvalue_allocator_tag {
     _Xlength_error("string too long");
 }
 
+extern "C" {
+void __sanitizer_annotate_contiguous_container(
+    const void* _First, const void* _End, const void* _Old_last, const void* _New_last) noexcept;
+extern const bool _Asan_string_should_annotate;
+}
+
+#pragma comment(linker, \
+    "/alternatename:__sanitizer_annotate_contiguous_container=__sanitizer_annotate_contiguous_container_default")
+#pragma comment(linker, "/alternatename:_Asan_string_should_annotate=_Asan_string_should_annotate_default")
+
+#ifdef __SANITIZE_ADDRESS__
+#pragma comment(lib, "stl_asan")
+#endif // __SANITIZE_ADDRESS__
+
+#if defined(__SANITIZE_ADDRESS__) || defined(_ANNOTATE_STRING)
+template <class _String, class = void>
+_INLINE_VAR constexpr bool _Has_minimum_allocation_alignment =
+    disjunction_v<bool_constant<alignof(typename _String::value_type) >= 8>,
+        is_same<typename _String::allocator_type, allocator<typename _String::value_type>>>;
+
+template <class _String>
+_INLINE_VAR constexpr bool _Has_minimum_allocation_alignment<_String,

```

Comment:
Is this name unique to `<string>` ?

Reply:
Urgh, now that one seems to be duplicated

Reply:
I believe we should move that one into `<xutility>`. It just happens to work because the vector PR has not been merged yet and I tried to keep them separate

Reply:
Ah, if they are exact duplicate it is no problem for them to be the same name